### PR TITLE
Keep language in logo link

### PR DESCRIPTION
**Pull request by chaconnewu**
This commit approaches the issue by clearing current search
states, cancelling any pending timeouts, and appending the
language to the url using window.history.pushState

### CODE REVIEW

File: static/js/index.js [hunk 1]
```diff
@@ -389,6 +394,21 @@
     newSearch(true, true);
   };
 
+  eltLogo.onclick = function() {
+    // Restore the state for "full" layout
+    setPageLayout("full");
+    eltSearchInput.value = "";
+
```
Comment (chaconnewu) [inline static/js/index.js]: Here I probably should clear the title as well

**Comment by sylvinus** [comment]
Hi @chaconnewu!

Thanks for the patch! 

However I think it's a bit too complex for what we want to achieve here. We are going to add more and more state into the page, and we'll need to clear that all just to make the logo link work again. Also transforming the `<a>` into a `<div>` is not good for javascript-less browsers.

Wouldn't it be more simple to just update the logo link dynamically and have the page reset entirely when the user follows the link?

File: static/js/index.js [hunk 2]
```diff
@@ -389,6 +394,21 @@
...
+    // Append the language info to the current url
+    window.history.pushState("", "", "?g=" + eltLang.value);
```
Comment (chaconnewu) [inline static/js/index.js]: Should add `if (window.history && window.history.pushState)`. If the browser does not support, can we `window.location.href="/"`?

**Comment by chaconnewu** [comment]
@sylvinus Thanks a lot for the comments and suggestions! I totally agree with you. What about the new implementation in the new commit? It pretty much follows what you described.

**Comment by sylvinus** [comment]
Great! Maybe we should do it in the `eltLang.onchange` instead?

**Comment by chaconnewu** [comment]
If we do it in `eltLang.onchange`, the language will be reset after page is reloaded, and you have to select the dropdown again in order retain the language postfix. Putting it in eltLogo will not have this issue. Or did I misunderstand the original issue?

**Comment by sylvinus** [comment]
Right - we should also change the link in the HTML template so that the right language is in the link when the page reloads?

**Comment by chaconnewu** [comment]
I used a function to set the href for logo when page load or language is selected with dropdown.

**Comment by sylvinus** [comment]
Great, thanks!

**Pull request merged**